Mathlib/MeasureTheory/Integral/RieszMarkovKakutani.lean

```diff
@@ -29,87 +34,425 @@ noncomputable section
 
 open BoundedContinuousFunction NNReal ENNReal
 
-open Set Function TopologicalSpace
+open Set Function TopologicalSpace ZeroAtInfty
+
+
+variable {X : Type*} [TopologicalSpace X] [LocallyCompactSpace X] [T2Space X] [NormalSpace X]
+variable (Λ : C(X, ℝ) →ₗ[ℝ] ℝ) (hΛ : ∀ (f : C(X, ℝ)), 0 ≤ f → 0 ≤ Λ f)
```

Comment:
It's not reasonable to start with a linear form defined on all continuous functions, because in most applications you won't have this. For instance the approach to define Lebesgue measure using Riesz-Markov-Kakutani is first to define a linear form on continuous compactly supported functions using the classical Riemann integral, and then apply the theorem to get Lebesgue (note that we won't use this approach as we already have Lebesgue measure in mathlib, this is just for illustrative purposes). But Riemann integral doesn't define a linear form on all continuous functions, only on compactly supported ones!

Reply:
Indeed, I agree with this, as there are *way* too many continuous functions on a locally compact space. @sgouezel we don't currently have a bundled version of the compactly supported continuous functions though, so what do you propose?

Reply:
yes, you are absolutely right! I wanted to change it to ` C₀(X, ℝ)` and forgot... I will try to do it. Or is it better to use compactly supported functions?

Reply:
One should indeed use compactly supported functions, because functions tending to zero at infinity are not always integrable -- so the Riemann integral for instance would not be well-defined on the function `1/(1+|x|)`, which tends to zero at infinity.

I think we should bundle compactly supported continuous functions to proceed there.

Reply:
the most general possible, sure!

I'm afraid I am not yet skilled enough to write new definitions, so it may take time to bundle compactly supported continuous functions. How should one proceed, is it a good idea to copy [ZeroAtInftyContinuousMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/ContinuousFunction/ZeroAtInfty.html#ZeroAtInftyContinuousMap), add say `has_compact_support'` and prove as many lemmas and coercions as possible?

Reply:
Exactly. You can copy the file to a new file `CompactlySupportedContinuousMap`, replace the field `zero_at_infty' with `HasCompactSupport toFun`, fix everything that breaks, and remove everything that doesn't make sense. This should be done in another PR, for easier reviewing. Don't hesitate to ask if you need help!

Reply:
I will try.

Should I replace `ZeroAtInfty` or extend it? I think it is convenient to have coercion from compactly supported functions to functions vanishing at infinity.

For the moment I use the symbol `C_c`, because apparently there is no small `c`. Is this fine?

Reply:
You should replace it. One can still register a coercion from compactly supported continuous functions to functions tending to zero at infinity even if it's not part of the definition -- this will require a proof, but that's normal. Otherwise, whenever someone defines a compactly supported functions, he would have to check both the compact support and the convergence to zero at infinity, which is a little bit silly.